# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.cpp
# Commit: f155dce29134
# Full Hash: f155dce2913463e41a589af8eec1dbc24c42aac3
# Author: Kashav Madan <kshvmdn@gmail.com>
# Date: 2021-06-16 09:46:08
# Regressor Bug: 1701337
# File Overlap Count: 1
# Description:
#   Bug 1701337 - Respect browser.sessionstore.debug.no_auto_updates in native flushers, r=nika
#   
#   We should not be doing any timed flushes if the pref is set to true.
#   
#   See https://searchfox.org/mozilla-central/rev/d97dca5907e2a0a77a2435b24ef980268cd3c4fe/browser/components/sessionstore/ContentSessionStore.jsm#402
# ==============================================================================

diff -r 14d2731278cb -r f155dce29134 docshell/base/CanonicalBrowsingContext.cpp
--- a/docshell/base/CanonicalBrowsingContext.cpp	Wed Jun 16 00:31:15 2021 +0000
+++ b/docshell/base/CanonicalBrowsingContext.cpp	Wed Jun 16 00:37:53 2021 +0000
@@ -2160,25 +2160,22 @@
     return;
   }
 
-  if (StaticPrefs::browser_sessionstore_debug_no_auto_updates()) {
-    UpdateSessionStoreSessionStorage([]() {});
-    return;
-  }
+  if (!StaticPrefs::browser_sessionstore_debug_no_auto_updates()) {
+    auto result = NS_NewTimerWithFuncCallback(
+        [](nsITimer*, void* aClosure) {
+          auto* context = static_cast<CanonicalBrowsingContext*>(aClosure);
+          context->UpdateSessionStoreSessionStorage([]() {});
+        },
+        this, StaticPrefs::browser_sessionstore_interval(),
+        nsITimer::TYPE_ONE_SHOT,
+        "CanonicalBrowsingContext::MaybeScheduleSessionStoreUpdate");
 
-  auto result = NS_NewTimerWithFuncCallback(
-      [](nsITimer*, void* aClosure) {
-        auto* context = static_cast<CanonicalBrowsingContext*>(aClosure);
-        context->UpdateSessionStoreSessionStorage([]() {});
-      },
-      this, StaticPrefs::browser_sessionstore_interval(),
-      nsITimer::TYPE_ONE_SHOT,
-      "CanonicalBrowsingContext::MaybeScheduleSessionStoreUpdate");
+    if (result.isErr()) {
+      return;
+    }
 
-  if (result.isErr()) {
-    return;
+    mSessionStoreSessionStorageUpdateTimer = result.unwrap();
   }
-
-  mSessionStoreSessionStorageUpdateTimer = result.unwrap();
 }
 
 void CanonicalBrowsingContext::CancelSessionStoreUpdate() {